### DISCUSSION TITLE
esp32-s3-4848S040 Relay GPIO pins?

### DISCUSSION THREAD

**Discussion by Tall-Paul** [HASwitchPlate/openHASP#681] [General]
Hi, I have this great 4 inch display which is supported in the nightly build (thank you eveyone involved in that).

I can't find anything about which pins the relays are on though.  I've tried pins 1 and 2 (there's some hints scattered about that these might be the right ones) but no joy.  Tried working backwards from 40 without much luck.  

Anyone know how to access the relays on this thing?  

Also, once I have GPIO pins configured in openhasp, do they pull through to HA as entities (using the openhasp plugin).

Cheers!

**Comment by fvanroie** [comment]
Someone needs to document this device. You can cross-reference https://github.com/HASwitchPlate/openHASP/issues/603 in the meantime.

**Reply by Tall-Paul** [reply]
I've been looking at that, it's how I first found out about openhasp (googling the device & 'home assistant' landed me on that issue)

However I don't believe the output pins are documented in there, unless they're 1,2 & 40...none of which seem to work for me.

**Reply by miwa4711** [reply]
Same here. Tried GPIOs 40, 41. Only GPIO40 works. Changed GPIO 1, 2 and 40 as power relays.  HA was still showing GPIO40 (ok) and 41 (but is not connected to relay as I learned). Solved it by delete plate in HA and added again. Nice!

**Comment by flexiti** [comment]
@Tall-Paul I am using #define MY_PK1 40   , and works

**Reply by Tall-Paul** [reply]
Ah Ha!

configuring pin 40 as a light Relay in openhasp, then sending {"state":"on"} to the hasp/plate001/command/40 topic worked.

Thank you!

**Reply by zneiva** [reply]
Hello,
how did you switch the relays?
I sent manual {"state": "on"} via MQTT, but nothing happened.
what do I have to set in openHASP and “plate”.

**Reply by palhaca** [reply]
same problem here!

**Reply by zneiva** [reply]
GPIO settings
Pin: "40" Type: "Light Relay" Group: "0" Default: "Normal"

I send hasp/plate/command/input40
{"state":"on"}

Answer

hasp/plate/state/input40 {"state":"off"}

**Reply by TNTLarsn** [reply]
hasp/plate/command/**output**40

It's a **output** pin

**Reply by zneiva** [reply]
thx TNTLarsn,
yes /output

**Reply by palhaca** [reply]
how can we implement that on the pages.jsonl file?

**Reply by palhaca** [reply]
ok i got it, here you have an example to local relay with the group id 1

{"page":5,"id":202,"obj":"btn","x":10,"y":40,"w":220,"h":115,"toggle":true,"text":"\uE335","text_font":32,"mode":"break","align":1,"radius":20,"groupid":1}

**Comment by Suschibomber** [comment]
Pinout for the 3 phases: GPIO1=L3, GPIO2=L2 und GPIO40=L1
